Sources/Holocron/FileSystemStorageProvider.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  FileSystemStorageProvider.swift
+//  Holocron-iOS
+//
+//  Created by Pranjal Satija on 4/15/19.
+//  Copyright © 2019 CocoaPods. All rights reserved.
+//
+
+import Foundation
+
+// MARK: Base Struct
+public extension StorageProvider {
+    typealias FileSystem = FileSystemStorageProvider
+}
+
+public struct FileSystemStorageProvider {
+    let baseURL: URL
+    let readTransformer: Transformer
+    let writeTransformer: Transformer
+    
+    init(baseURL: URL, readTransformer: @escaping Transformer = { $0 }, writeTransformer: @escaping Transformer = { $0 }) {
+        self.baseURL = baseURL
+        self.readTransformer = readTransformer
+        self.writeTransformer = writeTransformer
+    }
+    
+    func url(for key: String) -> URL {
+        return baseURL.appendingPathComponent(key)
+    }
+    
+    typealias Transformer = (Data) -> Data
+}
+
+// MARK: Public API
+extension FileSystemStorageProvider {
+    /// Writes a value for a specific key.
+    /// - parameter value: The value to store.
+    /// - parameter key: The key to associate with `value`.
+    /// - parameter options: The options to use when writing the file.
+    public func write<T: Encodable>(_ value: T, for key: String, options: Data.WritingOptions) throws {
```

Comment:
I'd like to see a `Key` type which is `ExpressibleByStringLiteral` and `RawRepresentable` so that for users who wish to do so, the writing keys can be a bit more type safe.

Reply:
@wmcginty I added this in the latest commit, but it's worth pointing out that `String` is **not** `RawRepresentable` by default for some odd reason, and that means that it can't be used as a key for `FileSystemStorageProvider` anymore without some sort of extension.

Reply:
The other storage providers, because they're implemented as extensions on existing types, which severely limits our abilities with generics, continue to use `String` as their key type, which I don't see as an issue because all the APIs we're abstracting away were meant to use `String` keys anyway.